internal/kafka/internal/config/node_prewarming_config.go

```diff
@@ -0,0 +1,101 @@
+package config
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/logger"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared"
+	"github.com/pkg/errors"
+)
+
+const defaultBaseStreamingUnitSize = "x1"
+
+type NodePrewarmingConfig struct {
+	filePath      string
+	Configuration map[string]InstanceTypeNodePrewarmingConfig
+}
+
+func NewNodePrewarmingConfig() NodePrewarmingConfig {
+	return NodePrewarmingConfig{
+		filePath:      "config/node-prewarming-configuration.yaml",
+		Configuration: make(map[string]InstanceTypeNodePrewarmingConfig),
+	}
+}
+
+func (c *NodePrewarmingConfig) ForInstanceType(instanceTypeID string) (InstanceTypeNodePrewarmingConfig, bool) {
+	instanceTypeConfig, found := c.Configuration[instanceTypeID]
+
+	if !found {
+		return InstanceTypeNodePrewarmingConfig{}, false
+	}
+
+	if instanceTypeConfig.BaseStreamingUnitSize == "" {
+		instanceTypeConfig.BaseStreamingUnitSize = defaultBaseStreamingUnitSize
+	}
+
+	return instanceTypeConfig, true
+}
+
+func (c *NodePrewarmingConfig) InstanceTypeConfigs() map[string]InstanceTypeNodePrewarmingConfig {
```

Comment:
The name seems a bit vague to me on what this method is for/doing. Maybe add `New` to it since it's creating a new config? Something like `NewInstanceTypeConfig()`?

Reply:
Thanks. I'll revisit this :-)

Reply:
This method is not used anywhere. I've removed it

Reply:
thanks for the follow up :)